test/integration/options/watch.spec.js

```diff
@@ -19,38 +21,156 @@ describe('--watch', function() {
       }
     });
 
-    it('should show the cursor and signal correct exit code, when watch process is terminated', function(done) {
-      this.timeout(0);
-      this.slow(3000);
-
-      var fixture = 'exit.fixture.js';
-      var spawnOpts = {stdio: 'pipe'};
-      var mocha = runMochaJSONRaw(
-        fixture,
-        args,
-        function postmortem(err, data) {
-          if (err) {
-            return done(err);
-          }
-
-          var expectedCloseCursor = '\u001b[?25h';
+    beforeEach(function() {
+      this.tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'mocha-'));
+
+      var fixtureSource = helpers.resolveFixturePath(helpers.DEFAULT_FIXTURE);
+
+      this.testFile = path.join(this.tempDir, 'test.js');
+      fs.copySync(fixtureSource, this.testFile);
+    });
+
+    afterEach(function() {
+      if (this.tempDir) {
+        return fs.remove(this.tempDir);
+      }
+    });
+
+    it('should show the cursor and signal correct exit code, when watch process is terminated', function() {
+      const [mocha, resultPromise] = runMochaJSONRawAsync(
+        helpers.DEFAULT_FIXTURE,
+        ['--watch']
+      );
+
+      return sleep(1000)
+        .then(() => {
+          mocha.kill('SIGINT');
+          return resultPromise;
+        })
+        .then(data => {
+          const expectedCloseCursor = '\u001b[?25h';
           expect(data.output, 'to contain', expectedCloseCursor);
 
-          function exitStatusBySignal(sig) {
-            return 128 + sig;
-          }
+          expect(data.code, 'to be', sigintExitCode);
+        });
+    });
+
+    it('reruns test when watched test file is touched', function() {
+      const [mocha, outputPromise] = runMochaJSONWatchAsync(this.testFile, [], {
+        cwd: this.tempDir
+      });
 
-          var sigint = 2;
-          expect(data.code, 'to be', exitStatusBySignal(sigint));
-          done();
-        },
-        spawnOpts
+      return sleep(1000)
```

Comment:
this is more idiomatic when using `unexpected`:

```js
return expect(sleep(1000).then(...), 'to have length', 2);
```

please change here & other places

Reply:
https://github.com/mochajs/mocha/pull/3915/commits/03049f6906d905c4cce2da2665c50e2299bfa840